docs/src/tutorial-nondeterministic-variables.md

```diff
@@ -41,26 +41,45 @@ The assertion we wrote in this harness was just an extra check we added to demon
 
 ## Custom nondeterministic types
 
-While `kani::any()` is the only method Kani provides to inject non-determinism into a proof harness, Kani only ships with implementations for a few types where we can guarantee safety.
-When you need nondeterministic variables of types that `kani::any()` cannot construct, you have two options:
-
-1. Implement the `kani::Arbitrary` trait for your type, so you can use `kani::any()`.
-2. Just write a function.
-
-The advantage of the first approach is that it's simple and conventional.
-It also means that in addition to being able to use `kani::any()` with your type, you can also use it with `Option<MyType>` (for example).
-
-The advantage of the second approach is that you're able to pass in parameters, like bounds on the size of the data structure.
+While `kani::any()` is the only method Kani provides to inject non-determinism into a proof harness, Kani only ships with implementations for a few `std` types where we can guarantee safety.
+When you need nondeterministic variables of types that don't have a `kani::any()` implementation available, you have the following options:
+
+1. Implement the `kani::Arbitrary` trait for your type, so you and downstream crates can use `kani::any()` with your type.
+2. Implement the [`bolero_engine::TypeGenerator` trait](https://docs.rs/bolero-engine/0.8.0/bolero_engine/trait.TypeGenerator.html).
```

Comment:
I think the best path would be using the `bolero_generator` crate directly. I don't think I meant to export it in the engine crate... But apparently I did 😁

https://docs.rs/bolero-generator/0.8.0/bolero_generator/derive.TypeGenerator.html

Reply:
That's funny, that's what I used initially.